scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+        return {"success": True, "content": str(content)}
+
+    def _make_fail(self, err):
+        return {"success": False, "content": str(err)}
+
+    def _load_config(self):
+        self._logger.info(f"Loading config from {CONFIG_FILE}")
+        try:
+            with open(CONFIG_FILE, 'r') as f:
+                self._config = json.load(f)
+            # JSON mandates str map keys, convert to int for convenience and consistency
+            self._config["nodes"] = {int(k): v for k, v in self._config["nodes"].items()}
+        except Exception as e:
+            self._error = f"Failed to load configuration from {CONFIG_FILE}: {str(e)}"
+            self._logger.error(self._error)
+            return self._make_error(self._error)
+        self._logger.debug(f"Loaded config: {json.dumps(self._config, indent=4)}")
+        return self._make_success()
+
+    def _get_node_config_item(self, i, key):
+        node_config = self._config["nodes"][i]
+        if key in node_config:
+            return node_config[key]
+
+        return self._config["default_scylla_config"][key]
+
+    def _send_reply(self, res = None):
+        if res is None:
+            res = self._make_success()
+        self._out.write(json.dumps(res) + '\n')
+        self._out.flush()
+
+    def start_cluster(self, **kwargs):
+        self._logger.info(f"start_cluster({kwargs})")
+        pass
+
+    def add_node(self, **kwargs):
+        self._logger.info(f"add_node({kwargs})")
+
+        i = self._get_next_index()
+        node_dir = f"node{i}"
+        os.mkdir(node_dir)
+
+        os.mkdir(os.path.join(node_dir, "data"))
+        os.mkdir(os.path.join(node_dir, "commitlog"))
+        os.mkdir(os.path.join(node_dir, "hints"))
+        os.mkdir(os.path.join(node_dir, "view_hints"))
+
+        node_ip = f"127.{self._config['network_prefix']}.0.{i}"
+
+        default_node_config = self._config["default_scylla_config"]
+        node_config = {
+            "index": i,
+            "ip": node_ip,
+        }
+        for item_name in default_node_config.keys():
+            value = kwargs.get(item_name, None)
+            if value is None or (type(value) == list and not value):
+                continue
+            node_config[item_name] = value
+
+        self._config["nodes"][i] = node_config
+
+        self._logger.debug(f"node_config is {node_config}")
+
+        scylla_path = self._get_node_config_item(i, "path")
+        scylla_mode = self._get_node_config_item(i, "mode")
+        shutil.copy(os.path.join(scylla_path, "build", scylla_mode, "scylla"), os.path.join(node_dir, "scylla"))
+        shutil.copy(os.path.join(scylla_path, "conf", "scylla.yaml"), os.path.join(node_dir, "scylla.yaml"))
+        shutil.copytree(os.path.join(scylla_path, "api", "api-doc"), os.path.join(node_dir, "api-doc"))
+
+        self._save_config()
+
+        return self._make_success(f"Added node with index {i}")
+
+    def _maybe_apply_all(self, callback, **kwargs):
```

Comment:
i'd suggest name the arguments instead of using keyword args everywhere. like:

```py
def _maybe_apply_all(self, callback, all=False, node_index=-1, /, **kwargs):
    if all and node_index >= 0:
        raise RuntimeError("Cannot pass both --node-index and --all to node operation")
    if not all and node_index < 0:
        raise RuntimeError("Either --node-index or --all has to be passed to node operation")
    # ...
```

Reply:
Done in v2.